client/data/authorizations/actions.ts

```diff
@@ -278,17 +293,31 @@ export function* submitCancelAuthorization(
 			)
 		);
 	} catch ( error ) {
+		const baseErrorMessage = sprintf(
+			// translators: %s Order id
+			__(
+				'There has been an error canceling the payment for order #%s.',
+				'woocommerce-payments'
+			),
+			orderId
+		);
+
+		const apiError = error as {
+			code?: string;
+			message?: string;
+			data?: {
+				status?: number;
+			};
+		};
+
+		const errorDetails =
+			apiError.message ||
```

Comment:
Are ther chances where the apiError message here leaks some internal context? Or is too generic? I feel we shouldn't concatenate the base message with the apiMessage directly, instead we can have some sort of mapping based on the error code to always be consistent with the error messages. WDYT?

Reply:
Good point! added the suggested mapping based on the API errors in this [commit]([8bfe52a](https://github.com/Automattic/woocommerce-payments/pull/9937/commits/8bfe52a49715b7df569b2fc3151d5d64e0294c2b)) thanks!